### ISSUE TITLE
Arbitrary key:value Opaque secrets not synced

### ISSUE THREAD

**Issue by megakoresh** [openshift/jenkins-sync-plugin#255]
If I define a secret like this:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: secret1
  labels:
    credential.sync.jenkins.openshift.io: 'true'
stringData:
  npmrc__auth: myamazingtokeneggsdedededede
  secret_fetish: too_weird_to_show
```

Then jenkins sync plugin will ignore it because it doesn't recognize any of it's usual patterns such as username/password or token and somesuch. It would be useful for these secrets to be still synced as Secret text type, especially if they are synced in Json format. Then pipeline script can parse the json into an object.

**Comment by bparees** [comment]
reopening to follow up on @megakoresh's comment in the PR about the type default to "opaque" which would send us through this path instead: https://github.com/gabemontero/jenkins-sync-plugin/blob/816dec56d9f89556a28a93f997a913b229f7c846/src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java#L215

which will not handle arbitrary keys, only specific keys.


> I don't see how this would fix #255
If no type is defined, the created secret will default to type Opaque (at least in Openshift 3.9), so this will never be executed. The default behaviour has to be also present for Opaque type (serialize to JSON and store as secret text). @gabemontero

**Comment by gabemontero** [comment]
v1.0.28 of the plugin has been initiated
